Lib/unittest/test/testmock/testpatch.py

```diff
@@ -1854,5 +1855,30 @@ def foo(*a, x=0):
             self.assertEqual(foo(), 1)
         self.assertEqual(foo(), 0)
 
+    def test_patch_dict_with_orderdict(self):
+        foo = OrderedDict()
+        foo['a'] = object()
+        foo['b'] = 'python'
+
+        original = foo.copy()
+        update_values = zip('cdefghijklmnopqrstuvwxyz', range(26))
+
+        @patch.dict(foo, OrderedDict(update_values))
```

Comment:
The assert looks incorrect that it's testing `list(foo) == sorted(foo)` but it should actually be testing whether `foo` contains `update_values` which on testing `foo['c']` gives `KeyError`. Sorry, I just noticed it now since the tests passed.

I checked this and `update_values` is a zip that becomes empty once it's iterated over and can be converted to a list to make sure that `update_values` can be used for subsequent calls. So how about below to make sure order is preserved?

```python
update_values = list(zip('cdefghijklmnopqrstuvwxyz', range(26)))
patched_items = list(foo.items()) + update_values

with patch.dict(foo, OrderedDict(update_values)):
    self.assertEqual(list(foo.items()), patched_items)
```

I would personally prefer a context manager here though other tests use `@patch.dict` with `test()` but that can wait for other reviewers.

Reply:
Hi @tirkarthi 

Yes I checked it manually and the foo['c'] gives the KeyError. I will change for the list(zip()) that you say. And I will use context manager, I think that is more readable.

Maybe, we could think about support convert zip to dict on OrderedDict()?

Reply:
> Maybe, we could think about support convert zip to dict on OrderedDict()?

Sorry, are you saying to use `patch.dict(foo, OrderedDict(dict(update_values)))` ? dict guarantees order currently and would help in catching the error if it changes in future. So I would prefer having `OrderedDict(update_values)`.

Reply:
I understand now.